linera-core/src/client/mod.rs

```diff
@@ -1727,23 +1727,12 @@ where
         }
         if let Some(cert) = info.manager.requested_locked {
             let hash = cert.hash();
-            let mut blobs = vec![];
-            while let Err(original_err) = self.client.handle_certificate(*cert.clone(), blobs).await
-            {
-                if let LocalNodeError::BlobsNotFound(blob_ids) = &original_err {
-                    if let Some(new_blobs) = remote_node
-                        .find_missing_blobs(blob_ids.clone(), chain_id)
-                        .await?
-                    {
-                        blobs = new_blobs;
-                        continue;
-                    }
-                }
+            let blobs = info.manager.locked_blobs.clone();
+            if let Err(err) = self.client.handle_certificate(*cert.clone(), blobs).await {
```

Comment:
Why do we no longer do the failover? i.e. finding the missing blobs and re-trying.

Reply:
Because we have at this point already received all the required blobs as part of the `ChainManagerInfo` anyway: The remote node has already sent them to us, and it was unnecessary in the first place to ask it again. So instead I'm taking the blobs out of the `ChainManagerInfo` and just process them now.
(Actually, we probably don't even need to clone, we can just take them…

Reply:
So b/c we now are sure all the required blobs are in `locked_blobs` we don't need the loop?

Reply:
Yes, exactly.